src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -17,132 +17,157 @@
-        } catch (JAXBException jaxbe) {
-            throw new StorageOperationException("Error parsing file data format");
-        } catch (IllegalValueException ive) {
-            throw new StorageOperationException("File contains illegal data values; data type constraints not met");
-        }
-    }
-
-    public String getPath() {
-        return path.toString();
-    }
+	/** Default file path used if the user doesn't provide the file name. */
+	public static final String DEFAULT_STORAGE_FILEPATH = "addressbook.txt";
+
+	
+
+	/*
+	 * Note: Note the use of nested classes below. More info
+	 * https://docs.oracle.com/javase/tutorial/java/javaOO/nested.html
+	 */
+
+	/**
+	 * Signals that the given file path does not fulfill the storage filepath
+	 * constraints.
+	 */
+	public static class InvalidStorageFilePathException extends IllegalValueException {
+		public InvalidStorageFilePathException(String message) {
+			super(message);
+		}
+	}
+
+	/**
+	 * Signals that some error has occured while trying to convert and
+	 * read/write data between the application and the storage file.
+	 */
+	public static class StorageOperationException extends Exception {
+		public StorageOperationException(String message) {
+			super(message);
+		}
+	}
+
+	private final JAXBContext jaxbContext;
+
+	public final Path path;
+	public boolean isFilesExisted;
+
+	/**
+	 * @throws InvalidStorageFilePathException
+	 *             if the default path is invalid
+	 */
+	public StorageFile() throws InvalidStorageFilePathException {
+		this(DEFAULT_STORAGE_FILEPATH);
+	}
+
+	/**
+	 * @throws InvalidStorageFilePathException
+	 *             if the given file path is invalid
+	 */
+	public StorageFile(String filePath) throws InvalidStorageFilePathException {
+		try {
+			jaxbContext = JAXBContext.newInstance(AdaptedAddressBook.class);
+		} catch (JAXBException jaxbe) {
+			throw new RuntimeException("jaxb initialisation error");
+		}
+
+		path = Paths.get(filePath);
+		if (!isValidPath(path)) {
+			throw new InvalidStorageFilePathException("Storage file should end with '.txt'");
+		}
+
+		isFilesExisted = CurrentFilePath();
+	}
+
+	/**
+	 * Returns true if the given path is acceptable as a storage file. The file
+	 * path is considered acceptable if it ends with '.txt'
+	 */
+	private static boolean isValidPath(Path filePath) {
+		return filePath.toString().endsWith(".txt");
+	}
+
+	/**
+	 * Saves all data to this storage file.
+	 *
+	 * @throws StorageOperationException
+	 *             if there were errors converting and/or storing data to file.
+	 */
+	public void save(AddressBook addressBook) throws StorageOperationException {
+
+		/*
+		 * Note: Note the 'try with resource' statement below. More info:
+		 * https://docs.oracle.com/javase/tutorial/essential/exceptions/
+		 * tryResourceClose.html
+		 */
+		if (!CurrentFilePath() && isFilesExisted) {
+			throw new StorageOperationException("File not found: " + path);
```

Comment:
Right place to check the existence of the file. Also consider creating the file and show the problem to the user, instead of closing the program.